tests/blackbox_test.go

```diff
@@ -311,3 +359,32 @@ func outer(t *testing.T, test types.Test, negativeTests bool) error {
 	}
 	return nil
 }
+
+// Identify and replace $uuid<num> with correct UUID
+// Variables with matching <num> should have identical UUIDs
+func replaceUUIDVars(t *testing.T, str string, UUIDmap map[int]string) (string, error) {
+	finalStr := str
+
+	pattern := regexp.MustCompile("\\$uuid([0-9]+)")
+	for _, match := range pattern.FindAllStringSubmatch(str, -1) {
+		if len(match) != 2 {
```

Comment:
I'd prefer to have it. Doesn't hurt and might prevent us from shooting outselves in the foot later on.

Reply:
Fair enough, this one I mostly brought up as if we end up switching the `UUIDmap` as mentioned in the other comment dropping this as well would allow us to no longer need to return an error in the `replaceUUIDVars` function.

Reply:
meh, error handling is fine.